torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -185,8 +185,7 @@ Tensor norm_backward(Tensor grad, const Tensor & self, const optional<Scalar> &
 }
 
 Tensor pow_backward(Tensor grad, const Tensor & self, const Scalar & exponent_) {
-  auto exponent = (exponent_.isComplex()) ? exponent_.toComplexDouble() : exponent_.toDouble();
-  if (exponent == 0.0) {
+  if ((exp.isComplex() && (exp.toComplexDouble() == 0.0)) || (!exp.isComplex() && (exp.toDouble() == 0.0))) {
```

Comment:
I assume this is written this way because "base == 0.0" doesn't work. Out of curiosity, what is the technical reason for why "base == 0.0" doesn't work?

Reply:
Nvm, it looks like Scalar doesn't have any math operators implemented on it (which is probably the right call)